### PR TITLE
Add moveable entity and do all required changes.

### PR DESCRIPTION
**Pull request by neumantm**

### CODE REVIEW

File: ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/entity/MovableEntity.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    
+    /**
+     * @return the current looking direction of this entity
+     */
+    public Direction getLookingDirection() {
+        return this.lookingDirection;
+    }
+    
+    private boolean isSolidEntityAt(final Position pos) {
+        return !this.getPlayfield().getEntitiesOfTypeAt(pos, SolidEntity.class, true).isEmpty();
```
Comment (buehlefs) [inline ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/entity/MovableEntity.java]: This should be a method of the playfield. And also the check for a class object is bad. Every entity should get a `isSolid() -> boolean` to indicate at runtime if it is solid or not. This is needed for the planned exit telephone.

File: ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/entity/SolidEntity.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * SPDX-License-Identifier:    MIT
+ */
+package de.unistuttgart.informatik.fius.icge.simulation.entity;
+
+/**
+ * A solid entity, through which a movable entity cannot move.
+ * 
+ * @author Tim Neumann
+ */
+public interface SolidEntity extends Entity {
```
Comment (buehlefs) [inline ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/entity/SolidEntity.java]: Solid state should not be part of the class hierarchy!

File: ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/internal/StandardSimulationClock.java [hunk 3]
```diff
@@ -104,6 +108,7 @@ private void tick() {
      *     The number of the simulation tick since the start of the clock.
      */
     private void tickSimulation(long tickNumber) {
+        System.out.println("TICK:" + tickNumber);
```
Comment (buehlefs) [inline ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/internal/StandardSimulationClock.java]: Remove debug output.

File: ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/SimulationClock.java [hunk 4]
```diff
@@ -86,4 +87,46 @@
...
+    long getLastTickNumber();
+    
+    /**
+     * Schedule an operation, to happen during the given tick. This method will block until that tick. Then the tick
+     * processing will halt until the given end of operation is completed.
+     * <p>
+     * If the given tick is in the past, the operation will be scheduled for the next tick.
+     * </p>
+     * 
+     * @param tick
```
[the remark this answers is not in the document]
Comment (neumantm) [reply ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/SimulationClock.java]: No this is not the duration. This this the absolute tick number when the operation will happen. I will improve the description.

[the remark this answers is not in the document]
Comment (neumantm) [reply ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/SimulationClock.java]: Also will rename the method to `scheduleOperationAtTick`

File: ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/entity/BasicEntity.java [hunk 5]
```diff
@@ -70,26 +71,47 @@ public void initOnPlayfield(Playfield playfield) {
     }
 
     /**
-     * Get the playfield of this entity.
-     * 
-     * @return the playfield
-     * @throws EntityNotOnFieldException
-     *     if this entity is not on a playfield
+     * @return whether this entity is on a playfield
```
[the remark this answers is not in the document]
Comment (neumantm) [reply ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/entity/BasicEntity.java]: ok

File: ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/entity/Direction.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * Copyright (c) 2019 the ICGE project authors.
+ * 
+ * This software is available under the MIT license.
+ * SPDX-License-Identifier:    MIT
+ */
+package de.unistuttgart.informatik.fius.icge.simulation.entity;
+
+/**
+ * The four fundamental directions possible in the simulation.
```
[the remark this answers is not in the document]
Comment (neumantm) [reply ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/entity/Direction.java]: ok

File: ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/entity/MovableEntity.java (hunk 1 above)
Comment (neumantm) [reply ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/entity/MovableEntity.java]: > This should be a method of the playfield. 

I agree.
>  Every entity should get a isSolid()

I'd like to get around this.
> This is needed for the planned exit telephone.

I will add a method to the interface, which returns the current solid state.

File: ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/entity/SolidEntity.java (hunk 2 above)
Comment (neumantm) [reply ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/entity/SolidEntity.java]: See comment above.

File: ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/internal/StandardSimulationClock.java (hunk 3 above)
Comment (neumantm) [reply ICGE-Simulation/src/main/java/de/unistuttgart/informatik/fius/icge/simulation/internal/StandardSimulationClock.java]: oh.

**Pull request merged**